pilot/pkg/serviceregistry/kube/controller.go

```diff
@@ -505,6 +506,10 @@ func (c *Controller) GetProxyServiceInstances(proxy *model.Proxy) ([]*model.Serv
 
 	pod := c.pods.getPodByIP(proxyIP)
 	if pod != nil {
+		// for k8s multi cluster, in case there are pods of the same ip across clusters
+		if !strings.HasPrefix(proxy.ID, pod.Name) || !strings.Contains(proxy.ID, pod.Namespace) {
```

Comment:
I guess that having two pods with the same name is pretty unlikely, but won't this check do the wrong thing in the corner case if proxy id is something like `xxxxx.myns-remote` and pod namespace is `myns`? Can't the check be more precise, e.g. something like: `if proxy.ID != pod.Name + "." + pod.Namespace`?

Reply:
Make sense